test/topology_tasks/test_tablet_tasks.py

```diff
@@ -280,3 +282,23 @@ async def check(type, log, mark):
     src = replicas[0].replicas[0]
     dst = (src[0], 1 - src[1])
     await asyncio.gather(move_tablet(src, dst), check("intranode_migration", log, mark))
+
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_tablet_task_sees_latest_state(manager: ManagerClient):
+    servers, cql, hosts, table_id = await create_table_insert_data_for_repair(manager)
+
+    token = -1
+    async def repair_task():
+        await inject_error_on(manager, "repair_tablet_fail_on_rpc_call", servers)
+        # Check failed repair request can be deleted
+        await manager.api.tablet_repair(servers[0].ip_addr, "test", "test", token)
+
+    async def del_repair_task():
+        tablet_task_id = empty_id
+        while tablet_task_id == empty_id:
+            tablet_task_id = await get_tablet_task_id(cql, hosts[0], table_id, token)
```

Comment:
get_tablet_task_id returns None or a valid task id. Why do we compare with 00000000-0000-0000-0000-000000000000?

Reply:
Oh. Looks like I broke the assumption. Fixing it.

Reply:
https://github.com/scylladb/scylladb/pull/22005